packages/react-core-auth/src/components/Authenticator/Machine/MachineContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from 'react';
+import { useInterpret, useSelector } from '@xstate/react';
+import {
+  AuthenticatorMachineOptions,
+  AuthMachineState,
+  getNextServiceFacade,
+  createAuthenticatorMachine,
+} from '@aws-amplify/ui';
+
+import { MachineContextType, UseMachineSelector, UseMachine } from './types';
+import { defaultComparator, getComparator } from './utils';
+
+export const USE_MACHINE_ERROR =
+  '`useMachine` must be used inside an `Authenticator.Provider`.';
```

Comment:
It shouldn't say to use it inside of MachineProvider?

Reply:
This is intentional, the customer would still be using `Authenticator.Provider` (can add a not explaining in a follow up)